src/lib.rs

```diff
@@ -109,22 +111,84 @@ const BACKEND_DIR: &str = "backend-auth";
 )]
 pub struct PinId(u8);
 
+/// Error obtained when trying to parse a [`PinId`][] either through [`PinId::from_path`][] or through the [`FromStr`](core::str::FromStr) implementation.
+#[derive(
+    Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Deserialize, Serialize,
+)]
+pub struct PinIdFromStrError;
+
 impl PinId {
-    fn path(&self) -> PathBuf {
+    /// Get the path to the PIN id.
+    ///
+    /// Path are of the form `pin.XX` where `xx` is the hexadecimal representation of the PIN number.
+    pub fn path(&self) -> PathBuf {
         let mut path = [0; 6];
         path[0..4].copy_from_slice(b"pin.");
         path[4..].copy_from_slice(&self.hex());
 
         PathBuf::from(&path)
     }
 
-    fn hex(&self) -> [u8; 2] {
+    /// Get the hex representation of the PIN id
+    pub fn hex(&self) -> [u8; 2] {
         const CHARS: &[u8; 16] = b"0123456789abcdef";
         [
             CHARS[usize::from(self.0 >> 4)],
             CHARS[usize::from(self.0 & 0xf)],
         ]
     }
+
+    /// Parse a PinId path
+    pub fn from_path(path: &str) -> Result<Self, PinIdFromStrError> {
+        let path = path
+            .strip_prefix("pin.")
+            .ok_or(PinIdFromStrError)?
+            .as_bytes();
+        if path.len() != 2 {
+            return Err(PinIdFromStrError);
+        }
+
+        let msb = match path[0] {
+            b'0' => 0x0,
+            b'1' => 0x1,
+            b'2' => 0x2,
+            b'3' => 0x3,
+            b'4' => 0x4,
+            b'5' => 0x5,
+            b'6' => 0x6,
+            b'7' => 0x7,
+            b'8' => 0x8,
+            b'9' => 0x9,
+            b'a' => 0xa,
+            b'b' => 0xb,
+            b'c' => 0xc,
+            b'd' => 0xd,
+            b'e' => 0xe,
+            b'f' => 0xf,
+            _ => return Err(PinIdFromStrError),
+        };
```

Comment:
If we are this verbose, can we use a helper function so that we only have to do it once?  Or maybe just use something like this:

```
fn parse_ascii_hex(a: u8) -> Option<u8> {
    match a {
        b'0'..=b'9' => Some(a - b'0'),
        b'a'..=b'f' => Some(a - b'a' + 0xA),
        _ => None,
    }
}
```

Reply:
This prompted me to look for a std solution and actually `u8::from_str_radix` is available in `core` so let's use that instead.